### ISSUE TITLE
Baseline series type

### ISSUE THREAD

**Issue by alekiv** [tradingview/lightweight-charts#151]
Is it possible to fill Area from predefined level (ex. 0)?
If you draw graph with negative values it currently fills area up from minimum value.
It maybe will be more logical to fill are from maximum to minimum.

https://jsfiddle.net/axtm3wr0/

![image](https://user-images.githubusercontent.com/9446663/61943551-5958cd00-afa4-11e9-8b93-2d2b64d89c8c.png)

**Comment by timocov** [comment]
It looks like Baseline chart type (you can see it on www.tradingview.com/chart).

**Comment by alekiv** [comment]
> It looks like Baseline chart type (you can see it on [www.tradingview.com/chart](http://www.tradingview.com/chart)).

Yep, it is.

**Comment by thanhlmm** [comment]
In fact, I'm working with this with the interface in like:
```js
basePrice: number;
underBaseTopColor: string;
underBaseBottomColor: string;
underLineColor: string;
```

It might not fit in future but feel free to give me suggetion

**Comment by timocov** [comment]
At the time in charting_library for baseline chart type we have the following properties, probably we can use some names/properties from them:

```
baselineColor // the color of baseline line
topFillColor1 // the start color of top area
topFillColor2 // the end color of top area 
bottomFillColor1 // the start color of bottom area
bottomFillColor2 // the end color of bottom area
topLineColor // the color of top line (above the baseline)
bottomLineColor // the color of bottom line (below the baseline)
topLineWidth // line width of top baseline
bottomLineWidth // line width of bottom baseline
baseLevelPercentage // percentage value of baseline, it might be from 0 to 100 and it is based on the current viewport, not on absolute chart value
```

**Comment by thanhlmm** [comment]
All make sense except `baseLevelPercentage` I think the base ine determined by price is more useful and configruable than level percentage

**Comment by timocov** [comment]
> All make sense except baseLevelPercentage I think the base ine determined by price is more useful and configruable than level percentage. So my suggestion is basePrice

@thanhlmm Yeah, probably we can make it configurable and do something like `baseValue: { type: 'price', price: number }` for now? In the future, if we'll decide to add new type of baseline value we can easily without a breaking change add it.

Also, I think it should be a different chart type with their own options (not extending the current one like line or area). Just to make it clearly and explicitly.

What do you think?

**Comment by thanhlmm** [comment]
Agree on both of your ideas

**Comment by timocov** [comment]
Closed in #813
